### PR TITLE
Remove sof-audio before scanning dmesg for error and Ignore known issues

### PR DESCRIPTION
**Pull request by fredoh9**
When scan errors in kernel log in sof-kernel-log-check.sh, current code filter "sof-audio" first then check for error. This will limit scanning logs and not scalable. Below $project_key is nothing but "sof-audio".

```
err_str="error|failed|timed out|panic|oops"
project_key="sof-audio"

if [ "$ignore_str" ]; then
    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str"|grep -vE "$ignore_str")
else
    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str")
fi
```

We should remove this limitation. If we see unrelated error, we can add it to ignoring error list. This is first step toward this approach. I ran first test on APL, CML Helios, ICL RVP. If no objection is raised, I will scan logs for all supported platforms and update ignore_str.

### CODE REVIEW

File: tools/sof-kernel-log-check.sh [hunk 1]
```diff
@@ -19,6 +18,12 @@ ignore_str="$ignore_str"'|error: status'
 # TODO explain
 ignore_str="$ignore_str"'|error: debugfs write failed to idle -16'
 
+# Realtek codecs thrown an error on startup, checking with Realtek
+ignore_str="$ignore_str"'|Parity error detected'
+
+# CML Helios known issue related with xhci_hcd
+ignore_str="$ignore_str"'|xhci_hcd 0000:00:14.0: WARN Set TR Deq Ptr cmd failed due to incorrect slot or ep state'
```
Comment (marc-hb) [inline tools/sof-kernel-log-check.sh:25]: Please one context URL (bug, mailing-list discussion, anything...) per ignored error. File a new bug/task if really needed.

File: tools/sof-kernel-log-check.sh [hunk 2]
```diff
@@ -35,9 +40,9 @@ ignore_str="$ignore_str"'|error: debugfs write failed to idle -16'
 #echo "run $0 with parameter '$*' for check kernel message error"
 
 if [ "$ignore_str" ]; then
-    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str"|grep -vE "$ignore_str")
+    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep -E "$err_str"|grep -vE "$ignore_str")
 else
-    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep $project_key | grep -E "$err_str")
+    err=$(eval $cmd|grep 'Call Trace' -A5 -B3)$(eval $cmd | grep -E "$err_str")
```
Comment (marc-hb) [inline tools/sof-kernel-log-check.sh:45]: The `eval` seems unnecessary, can you try to remove it?

Comment (fredoh9) [reply tools/sof-kernel-log-check.sh:47]: Without it, I'm seeing errors.  How about handling this when we switch to journalctl
```
sed: -e expression #1, char 1: unknown command: `''
sed: -e expression #1, char 1: unknown command: `''

```

**Comment by fredoh9** [comment]
> > err_str="error|failed|timed out|panic|oops"
> 
> Later we should drop such "screen scraping" and use something more robust like `journalctl --dmesg --priority err` with no false positive (for instance a debug message with one of these keywords) and no false negative (e.g. `ERR:`). I'm not suggesting to do that right now but probably good to keep in mind while you're doing this PR.

This is good idea, I will create an issue to follow up.

File: tools/sof-kernel-log-check.sh (hunk 2 above)
Comment (marc-hb) [reply tools/sof-kernel-log-check.sh:47]: OK forget it for now; one change at a time.

**Pull request merged**